pypesto/visualize/model_fit.py

```diff
@@ -174,24 +176,72 @@ def time_trajectory_model(
     if timepoints is None:
         end_time = max(problem.objective.edatas[0].getTimepoints())
         timepoints = np.linspace(start=0, stop=end_time, num=n_timepoints)
-    obj = problem.objective.set_custom_timepoints(timepoints_global=timepoints)
 
-    # evaluate objective with return dic = True to get data
+    # get optimization result
     parameters = result.optimize_result.list[start_index]['x']
+
     # reduce vector to only include free indices. Needed downstream.
     parameters = problem.get_reduced_vector(parameters)
-    ret = obj(parameters, mode='mode_fun', sensi_orders=(0,), return_dict=True)
+
+    # simulate with custom timepoints for hierarchical model
+    if isinstance(problem.objective.calculator, HierarchicalAmiciCalculator):
```

Comment:
could also outsource everything under this condition to a separate helper function that retrieves the rdatas

Reply:
Liked that idea, so it is also usable for other functions or to the users. Put a function for that into the `misc.py`.